src/bundle/Resources/public/js/scripts/core/dropdown.js

```diff
@@ -478,6 +476,31 @@
                 .forEach((option) => option.addEventListener('click', this.onOptionClick, false));
 
             if (this.itemsFilterInput) {
+                const modal = this.container.closest('.modal');
+                const popupInputs = this.itemsContainer.querySelectorAll('input');
+
+                popupInputs.forEach((popupInput) =>
+                    popupInput.addEventListener(
+                        'focusin',
+                        () => {
+                            const modalInstance = bootstrap.Modal.getInstance(modal);
+
+                            if (modalInstance) {
+                                modalInstance._focustrap.deactivate();
```

Comment:
Bootstrap modal uses FocusTrap which steals focus if it is outside the modal
https://github.com/twbs/bootstrap/blob/597c4023141dc48868889b85676b2d7269411d23/js/src/modal.js#L163